src/app/proposals/create/TreasuryWithdrawProposalForm.tsx

```diff
@@ -35,37 +35,45 @@ import { useForm } from 'react-hook-form'
 import { CreateProposalHeaderSection } from '@/app/proposals/create/CreateProposalHeaderSection'
 import { isAddressRegex } from '@/app/proposals/shared/utils'
 import { isBaseError, isUserRejectedTxError } from '@/components/ErrorPage/commonErrors'
+import { useAccount } from 'wagmi'
 
 const rifMinimumAmount = ENV === 'mainnet' ? 10 : 1
 const rbtcMinimumAmount = ENV === 'mainnet' ? 0.0001 : 0.000001
 
-const FormSchema = z
-  .object({
-    proposalName: z
-      .string()
-      .max(100)
-      .refine(s => s.trim().replace(/\s+/g, ' ').length >= 5, 'Field must contain at least 5 characters'),
-    description: z
-      .string()
-      .max(3000)
-      .refine(s => s.trim().replace(/\s+/g, ' ').length >= 10, 'Field must contain at least 10 characters'),
-    toAddress: z.string().refine(value => isAddressRegex(value), 'Please enter a valid address'),
-    tokenAddress: z.string().length(42),
-    amount: z.coerce
-      .number({ invalid_type_error: 'Required field' })
-      .positive('Required field')
-      .max(MAX_INPUT_NUMBER_AMOUNT),
-  })
-  .refine(data => data.tokenAddress === zeroAddress || Number(data.amount) >= rifMinimumAmount, {
-    message: `The minimum amount for ERC20 tokens is ${rifMinimumAmount}.0`,
-    path: ['amount'],
-  })
-  .refine(data => data.tokenAddress !== zeroAddress || Number(data.amount) >= rbtcMinimumAmount, {
-    message: `The minimum amount for RBTC is ${rbtcMinimumAmount}`,
-    path: ['amount'],
-  })
-
 export const TreasuryWithdrawProposalForm = () => {
+  const { chainId } = useAccount()
```

Comment:
I am seeing this and something pops to my head: why do we need to move this logic within React ecosystem and not keep it outside?

Reply:
It was required just for chainId. I've updated to use CHAIN_ID constant, so I return the FormSchema outside the react.